### DISCUSSION TITLE
Some format not work in django rest framework

### DISCUSSION THREAD

**Discussion by Chhunneng** [ImageMagick/ImageMagick#6445] [Development]
### ImageMagick version

6.9.12-82

### Operating system, version and so on

Amazon Linux 2023

### Description

As I try to convert many file type on my computer, it worked normally. But in django rest framework on webserver, not work. In serializer, I create a temporary directory, then I save file from request body to directory and I use wand to read image from temporary and write it back for converted file. It shows error and not sure it error because wand or imagemagick
```bash
sh: line 1: JxrDecApp: command not found
/usr/bin/mv: cannot stat '/tmp/magick-tmdXaDsLOZqhpZw5bieSNrektDXDvS3P.pnm': No such file or directory
Wand Error: unable to open file `/tmp/magick-Ki9jITFD8z4uhvzl22oySVB5TpnAlMon': No such file or directory @ error/constitute.c/ReadImage/620
```

Here is my code

```python
temp_dir = tempfile.TemporaryDirectory()
original_picture = validated_data.pop("picture")
temp_original_picture_path = os.path.join(temp_dir.name, "filename.ARW")
temp_converted_picture_path = os.path.join(temp_dir.name, "filename.png")
# store file in TemporaryDirectory
with open(temp_original_picture_path, 'wb') as f:
    for chunk in original_picture.chunks():
        f.write(chunk)
try:
    from wand.image import Image as WandImage
    with WandImage(filename=temp_original_picture_path) as image:
        image.format = 'png'
        image.save(filename=temp_converted_picture_path)
except Exception as e:
    print("Wand Error:", e)
```
Note: but when I try on my PC it works normal

**Comment by snibgo** [comment]
> sh: line 1: JxrDecApp: command not found

`delegates.xml` in your ImageMagick directory probably contains a line something like this (Windows version):
```
  <delegate decode="jxr" command="cmd.exe /c (move &quot;%i&quot; &quot;%i.jxr&quot; >nul) &amp; (&quot;JXRDecApp.exe&quot; -i &quot;%i.jxr&quot; -o &quot;%o.pnm&quot;) &amp; (move &quot;%i.jxr&quot; &quot;%i&quot; >nul) &amp; (move &quot;%o.pnm&quot; &quot;%o&quot; >nul)"/>
```
So your code is trying to read a JXR file, by delegating the work to the program `JXRDecApp`. The error is saying that you don't have that program.

**Reply by Chhunneng** [reply]
How can I have JXRDecApp? as I try to install on Amazon Linux 2023 it can not install.

**Reply by fmw42** [reply]
What does this have to do with Imagemagick?

**Reply by Chhunneng** [reply]
wand.image using Imagemagick

**Reply by fmw42** [reply]
There is no such app in Imagemagick thus not in Wand.  Please give us a reference/link to understand better.

**Reply by Chhunneng** [reply]
https://docs.wand-py.org/en/0.6.11/
In code it require MagickWand

**Reply by fmw42** [reply]
What does JxrDecApp have to do with Python Wand?   What is JxrDecApp?  Do you have a reference to that directly?

If you think it is relevant to Python Wand, then open an issue at https://github.com/emcconville/wand

**Comment by snibgo** [comment]
> How can I have JXRDecApp?

I don't know. That program is not part of the ImageMagick system.

**Comment by fmw42** [comment]
JxrDecApp  is a C decoder app for JXR format.  See https://manpages.ubuntu.com/manpages/jammy/man1/JxrDecApp.1.html. That app has nothing to do with Imagemagick.  

Do you want JXR conversion in Imagemagick?  Is that a request for a new format? If so, it is discussed at https://www.loc.gov/preservation/digital/formats/fdd/fdd000524.shtml#:~:text=JPEG%20XR%20(JPEG%20extended%20range,(formerly%20Windows%20Media%20Photo). 

JXRlib is used for reading and writing that format. See https://github.com/curasystems/jxrlib/tree/master.  

However, I do not see JXR as implemented in Imagemagick. It is not listed for the available formats currently as far as I can tell.  See https://imagemagick.org/script/formats.php

If it is not in Imagemagick, it won't be available in Python Wand.
